### PR TITLE
 CHANGELOG: RAC 4 section

### LINKED ISSUE #2682: RAC4 Backwards compatibility with RAC2.X
Hi!

We are still on a mainly Objective-C and have been waiting for RAC4 to become official to start using it with Swift. 

I'm wondering if this, from the RAC3.0 release notes is still true in RAC4.

> ReactiveCocoa 3.0 includes the first official Swift API, which is intended to eventually supplant the Objective-C API entirely.
>
>However, because migration is hard and time-consuming, and because Objective-C is still in widespread use, 99% of RAC 2.x code will continue to work under RAC 3.0 without any changes.

If so it might be nice to add to the release notes. Looking forward to using the new version, you guys do great work! :+1:

### PR DESCRIPTION
**Pull request by NachoSoto**
Completed `CHANGELOG` for `RAC 4`. Instead of adding on top of the existing `alpha` sections I've actually rearranged everything, since most people won't care about the process but the combined sets of changes since `RAC 3`.

It might be easier to review this by actually opening the file rather than looking at the diff.

cc @ReactiveCocoa/ReactiveCocoa.

Once this is landed, I'll publish the drafted `4.0` release! :tada:
Fixes #2682.

### CODE REVIEW

**Comment by NachoSoto** [comment]
Let me know what you all think! I'll merge this in a few hours unless I hear otherwise :speak_no_evil:

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,146 +1,113 @@
...
+#### Renamed `try` and `catch` operators
 
-#### Renamed PropertyOf<T> to AnyProperty<T>
+The `try` and `catch` operators were renamed because of the addition of the error handling keywords with the same name. They are now `attempt` and `flatMapError` respectively. Also, `tryMap` was renamed to `attemptMap` for consistency.
 
-This is in-line with changes to the standard library in Swift 2.
+#### `flatten` and `flatMap` are now possible for all 4 combinations of `Signal`+`SignalProducer`
 
-#### Publicized Bag and Atomic
+This fills a gap that was missing in `RAC 3`. It’s a common pattern to have a hot `Signal` of values that need to be mapped to “work” — `SignalProducer`, or a `Signal` of `Signal`s.
```
Comment (rpowelll) [inline CHANGELOG.md]: This reads a little awkwardly to me, it took me rereading it a few times to get what it was trying to say,

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,146 +1,113 @@
...
-The improvements to associated enum values in Swift 2 mean that `Event` cases
-no longer need to be `Box`ed. In fact, the `Box` dependency has been removed
-completely from RAC 4.
+#### Publicized `Bag` and `Atomic`
 
-#### Replacements for the start and observer overloads
+`Bag` and `Atomic` are now public. These are useful when creating custom operators for RAC types.
 
-_These are likely to see further changes in a later alpha_
+#### `SignalProducer.buffer` no longer has a default value
```
Comment (bencochran) [inline CHANGELOG.md]: Maybe “no longer has a default capacity”? “value” is rather overloaded.

File: CHANGELOG.md (hunk 1 above)
Comment (eimantas) [inline CHANGELOG.md]: Ditto. Still can't wrap my head around this sentence.

**Comment by mdiep** [comment]
It looks good to me with the caveat that we should make the issue references into links. (This doesn't happen automatically for files in a repository—only for content on GitHub.com The Website.)

Thanks for tackling this! :sparkles:

File: CHANGELOG.md (hunk 1 above)
Comment (NachoSoto) [inline CHANGELOG.md]: Thanks! updated.

File: CHANGELOG.md (hunk 2 above)
Comment (NachoSoto) [inline CHANGELOG.md]: Good call!

**Comment by NachoSoto** [comment]
Thanks @mdiep! I wasn't aware of that, I figured GitHub would display links automatically once merged.

I think I've addressed all the comments :)

**Comment by bencochran** [comment]
Looks great to me!

**Comment by mdiep** [comment]
:ship:

**Comment by NachoSoto** [comment]
:heart_eyes: any more thoughts?

**Comment by neilpa** [comment]
:shipit:

**Comment by NachoSoto** [comment]
@bencochran, @mdiep, and @neilpa approves. I'm happy :walking:

**Comment by NachoSoto** [comment]
no pressure

**Comment by NachoSoto** [comment]
All right, let's do this!
mission accomplished

**Pull request merged**